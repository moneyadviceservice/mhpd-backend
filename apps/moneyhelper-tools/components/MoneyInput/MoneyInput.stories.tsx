import { StoryFn } from '@storybook/react';
import { MoneyInput } from './MoneyInput';

const StoryProps = {
  title: 'Components/TOOLS/MoneyInput',
  component: MoneyInput,
};

const Template: StoryFn = (args) => <MoneyInput {...args} />;

export const Default = Template.bind({});
Default.args = {
  label: 'label',
  id: 'id',
  name: 'name',
  defaultValue: '',
  onChange: () => {},
};

export default StoryProps;
