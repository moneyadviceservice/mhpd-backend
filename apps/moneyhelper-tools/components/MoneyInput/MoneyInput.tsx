import type { Props as NumberInputProps } from '../NumberInput';
import { NumberInput } from '../NumberInput';
import { twMerge } from 'tailwind-merge';
export type MoneyInputProps = NumberInputProps;

type Props = MoneyInputProps & {
  inputClassName?: string;
  addon?: string;
  inputBackground?: string;
  handleErrorClick?: boolean;
};

export const MoneyInput = ({
  thousandSeparator = true,
  inputClassName,
  addon,
  inputBackground,
  handleErrorClick,
  ...props
}: Props) => {
  return (
    <div className={twMerge('flex', inputClassName)}>
      <span
        aria-hidden
        className="bg-gray-100 py-2 px-3 rounded-l border-gray-400 border border-solid leading-[31px] text-[18px]"
      >
        Â£
      </span>
      <NumberInput
        thousandSeparator={thousandSeparator}
        className={twMerge(
          'border-gray-400 border-t border-b border-r border-l-0',
          addon ? 'rounded-none border-r-0' : 'rounded-r rounded-l-none',
          inputBackground,
        )}
        {...props}
      />
      {addon && (
        <span
          aria-hidden
          className="bg-gray-100 py-2 px-3 rounded-r border-gray-400 border border-solid text-nowrap"
        >
          {addon}
        </span>
      )}
    </div>
  );
};
