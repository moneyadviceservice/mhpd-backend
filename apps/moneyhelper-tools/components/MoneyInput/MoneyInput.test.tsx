import { render } from '@testing-library/react';
import { MoneyInput } from './MoneyInput';

describe('MoneyInput component', () => {
  it('renders correctly', () => {
    const { container } = render(
      <MoneyInput
        id="test-id"
        name="test-name"
        defaultValue=""
        onChange={() => {}}
      />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders correctly with addon', () => {
    const { container } = render(
      <MoneyInput
        id="test-id"
        name="test-name"
        defaultValue=""
        onChange={() => {}}
        addon="Â£"
      />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
