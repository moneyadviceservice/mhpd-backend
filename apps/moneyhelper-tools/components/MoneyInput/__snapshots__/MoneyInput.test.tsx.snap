// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoneyInput component renders correctly 1`] = `
<div
  class="flex"
>
  <span
    aria-hidden="true"
    class="bg-gray-100 py-2 px-3 rounded-l border-gray-400 border border-solid leading-[31px] text-[18px]"
  >
    £
  </span>
  <input
    class="tool-field w-full border pl-2 ml-0 h-auto focus:outline-purple-700 focus:shadow-focus-outline border-gray-400 border-t border-b border-r border-l-0 rounded-r rounded-l-none"
    data-testid="number-input"
    id="test-id"
    inputmode="numeric"
    name="test-name"
    type="text"
    value=""
  />
</div>
`;

exports[`MoneyInput component renders correctly with addon 1`] = `
<div
  class="flex"
>
  <span
    aria-hidden="true"
    class="bg-gray-100 py-2 px-3 rounded-l border-gray-400 border border-solid leading-[31px] text-[18px]"
  >
    £
  </span>
  <input
    class="tool-field w-full border pl-2 ml-0 h-auto focus:outline-purple-700 focus:shadow-focus-outline border-gray-400 border-t border-b border-l-0 rounded-none border-r-0"
    data-testid="number-input"
    id="test-id"
    inputmode="numeric"
    name="test-name"
    type="text"
    value=""
  />
  <span
    aria-hidden="true"
    class="bg-gray-100 py-2 px-3 rounded-r border-gray-400 border border-solid text-nowrap"
  >
    £
  </span>
</div>
`;
